tools/serverpod_cli/test_e2e/test/generate_watch_test.dart

```diff
@@ -0,0 +1,623 @@
+@Timeout(Duration(minutes: 8))
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:collection/collection.dart';
+import 'package:path/path.dart' as path;
+import 'package:serverpod_cli_e2e_test/src/keyword_search_in_stream.dart';
+import 'package:test/test.dart';
+import 'package:uuid/uuid.dart';
+
+const generateWatchCompletionKeywords = [
+  'Initial code generation complete. Listening for changes.',
+  'Incremental code generation complete.',
+];
+
+void main() async {
+  var tempPath = path.join(Directory.current.path, 'temp');
+  var rootPath = path.join(Directory.current.path, '..', '..', '..');
+  var cliPath = path.join(rootPath, 'tools', 'serverpod_cli');
+
+  setUpAll(() async {
+    await Process.run(
+      'dart',
+      ['pub', 'global', 'activate', '-s', 'path', '.'],
+      workingDirectory: cliPath,
+    );
+
+    Directory(tempPath).createSync(recursive: true);
+  });
+
+  tearDownAll(() async {
+    Directory(tempPath).deleteSync(recursive: true);
+  });
+
+  group('Given protocol file that is changed when generate watch is active',
+      () {
+    var (projectName, commandRoot) = createRandomProjectName(tempPath);
+    var (serverDir, _, clientDir) = createProjectFolderPaths(projectName);
+
+    late Process createProcess;
+    Process? generateWatch;
+    KeywordSearchInStream? generateStreamSearch;
+    tearDown(() async {
+      createProcess.kill();
+      generateWatch?.kill();
+      generateStreamSearch?.close();
+
+      await Process.run(
+        'docker',
+        ['compose', 'down', '-v'],
+        workingDirectory: commandRoot,
+      );
+
+      while (!await isNetworkPortAvailable(8090)) {}
+    });
+
+    test('then entity files are generated and updated as expected.', () async {
+      // Create project
+      createProcess = await Process.start(
+        'serverpod',
+        ['create', projectName, '-v', '--no-analytics'],
+        workingDirectory: tempPath,
+        environment: {
+          'SERVERPOD_HOME': rootPath,
+        },
+      );
+
+      createProcess.stdout.transform(const Utf8Decoder()).listen(print);
+      createProcess.stderr.transform(const Utf8Decoder()).listen(print);
+
+      var createProjectExitCode = await createProcess.exitCode;
+      expect(
+        createProjectExitCode,
+        0,
+        reason: 'Failed to create the serverpod project.',
+      );
```

Comment:
Question: Shouldn't this be part of the setup? As we are not testing the project creation.

Reply:
Yes, good catch! We don't actually need this in the test.